### PR TITLE
chore: add utility methods for checking docker container state

### PR DESCRIPTION
**Pull request by KollaAdithya**
This is part 2 of implementation of container dependencies 
adding utility methods for checking different states of a container.
In the next PR will be to call these methods based of dependencies of each container in orchestrator pkg.

### CODE REVIEW

**Review by CaptainCarpensir — APPROVED** [review]
Changes are extremely easy to understand, looks great!

File: internal/pkg/docker/dockerengine/dockerengine.go [hunk 1]
```diff
@@ -339,13 +353,103 @@ func (c DockerCmdClient) Run(ctx context.Context, options *RunOptions) error {
...
+	case containerStatusRunning:
+		return true, nil
+	case containerStatusExited:
+		return false, &ErrContainerExited{name: name, exitcode: state.ExitCode}
+	}
+	return false, nil
+}
+
+// IsContainerCompleteOrSuccess returns true if a docker container exits with an exitcode.
+func (c DockerCmdClient) IsContainerCompleteOrSuccess(ctx context.Context, containerName string) (bool, int, error) {
```
Comment (iamhopaul123) [inline internal/pkg/docker/dockerengine/dockerengine.go]: Is it a good idea to return `-1` when the container is still running 💭 And the method just returns the exit code.

File: internal/pkg/docker/dockerengine/dockerengine.go [hunk 2]
```diff
@@ -339,13 +353,103 @@ func (c DockerCmdClient) Run(ctx context.Context, options *RunOptions) error {
...
+	}
+	switch state.Health.Status {
+	case healthy:
+		return true, nil
+	case starting:
+		return false, nil
+	case unhealthy:
+		return false, fmt.Errorf("container %q is %q", containerName, unhealthy)
+	case noHealthcheck:
+		return false, fmt.Errorf("healthcheck configuration is set to %q for container %s", noHealthcheck, containerName)
```
Comment (iamhopaul123) [inline internal/pkg/docker/dockerengine/dockerengine.go]: We should probably return the same error as https://github.com/aws/copilot-cli/pull/5488/files#diff-245a109ace8ab38c02069b4557e8256a2fcf479ab23323d871df8357ecce6b51R388

File: internal/pkg/docker/dockerengine/dockerengine.go [hunk 3]
```diff
@@ -339,13 +353,103 @@ func (c DockerCmdClient) Run(ctx context.Context, options *RunOptions) error {
...
+	return state.Status == containerStatusExited, state.ExitCode, nil
+}
+
+// IsContainerHealthy returns true if a container health state is healthy.
+func (c DockerCmdClient) IsContainerHealthy(ctx context.Context, containerName string) (bool, error) {
+	state, err := c.containerState(ctx, containerName)
+	if err != nil {
+		return false, err
+	}
+	if state.Status == containerStatusExited {
```
Comment (iamhopaul123) [inline internal/pkg/docker/dockerengine/dockerengine.go]: maybe check if the status is "running" instead? and error out if the container is not running.

**Review by iamhopaul123 — APPROVED** [review]
:shipit:

**Pull request merged**